Formula/avr-gcc@11.rb

```diff
@@ -132,22 +138,20 @@ def install
       ENV.delete "CC"
       ENV.delete "CXX"
 
+      # avr-libc ships with outdated config.guess and config.sub scripts that
+      # do not support Apple ARM systems, causing the configure script to fail.
       build_config = `./config.guess`.chomp
+      if build_config.start_with?("-")
```

Comment:
why `"-"`?

Reply:
I saw that config.guess returned `-apple-darwin20.1.0` and then config.sub had an error because that's an invalid [triplet](https://wiki.osdev.org/Target_Triplet).   Obviously I didn't want to hardcode that string but I wanted to come up with some simple way to detect the problem.

Any triplet that starts with `-` is probably wrong, because the triple should start with the name of the machine that is being targeted.  So it seemed fine to just use that as the condition.  

I couldn't figure out how to ask Homebrew whether we are building on an M1 machine.

Reply:
You can use `Hardware::CPU.arm?`, it would be better than the triplet check.

See https://github.com/Homebrew/homebrew-core/blob/master/Formula/gcc.rb#L74